File: web/src/api/competitionTeam.js
import service from '@/utils/request'

export const createCompetitionTeam = (data) => {
  return service({
    url: '/competitionTeam/competitionTeam',
    method: 'post',
    data
  })
}

export const updateCompetitionTeam = (data) => {
  return service({
    url: '/competitionTeam/competitionTeam',
    method: 'put',
    data
  })
}

export const deleteCompetitionTeam = (data) => {
  return service({
    url: '/competitionTeam/competitionTeam',
    method: 'delete',
    data
  })
}

export const getCompetitionTeam = (params) => {
  return service({
    url: '/competitionTeam/competitionTeam',
    method: 'get',
    params
  })
}

export const getCompetitionTeamList = (params) => {
  return service({
    url: '/competitionTeam/competitionTeamList',
    method: 'get',
    params
  })
}

export const importExcel = (file, mode) => {
  const formData = new FormData()
  formData.append('file', file)
  return service({
    url: `/competitionTeam/importExcel?mode=${mode}`,
    method: 'post',
    data: formData,
    headers: { 'Content-Type': 'multipart/form-data' }
  })
}

export const addWarID = (data) => {
  return service({
    url: '/competitionTeam/addWarID',
    method: 'post',
    data
  })
}

export const getTeamScores = (params) => {
  return service({
    url: '/competitionTeam/scores',
    method: 'get',
    params
  })
}

export const getTeamRecentScores = (params) => {
  return service({
    url: '/competitionTeam/recentScores',
    method: 'get',
    params
  })
}

export const getTeamDetail = (params) => {
  return service({
    url: '/competitionTeam/detail',
    method: 'get',
    params
  })
}

export const getAllTeamsScoreSummary = () => {
  return service({
    url: '/competitionTeam/allScores',
    method: 'get'
  })
}

export const deleteTeamScore = (params) => {
  return service({
    url: '/competitionTeam/deleteScore',
    method: 'delete',
    params
  })
}

export const updateTeamScore = (data) => {
  return service({
    url: '/competitionTeam/updateScore',
    method: 'put',
    data
  })
}

export const calculateWarIDForAllTeams = (data) => {
  return service({
    url: '/competitionTeam/calculateWarID',
    method: 'post',
    data
  })
}

export const confirmWarIDScores = (data) => {
  return service({
    url: '/competitionTeam/confirmWarID',
    method: 'post',
    data
  })
}

// 公开接口（无需鉴权）
export const getPublicTeamList = () => {
  return service({
    url: '/competitionTeam/public/teamList',
    method: 'get'
  })
}

File: server/router/example/competition_team.go
package example

import (
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type CompetitionTeamRouter struct{}

func (c *CompetitionTeamRouter) InitCompetitionTeamRouter(Router, PublicRouter *gin.RouterGroup) {
	router := Router.Group("competitionTeam").Use(middleware.OperationRecord())
	routerWithoutRecord := Router.Group("competitionTeam")
	{
		router.POST("competitionTeam", competitionTeamApi.CreateCompetitionTeam)
		router.PUT("competitionTeam", competitionTeamApi.UpdateCompetitionTeam)
		router.DELETE("competitionTeam", competitionTeamApi.DeleteCompetitionTeam)
		router.POST("importExcel", competitionTeamApi.ImportExcel)
		router.POST("addWarID", competitionTeamApi.AddWarID)
		router.POST("calculateWarID", competitionTeamApi.CalculateWarIDForAllTeams)
		router.POST("confirmWarID", competitionTeamApi.ConfirmWarIDScores)
		router.PUT("updateScore", competitionTeamApi.UpdateTeamScore)
		router.DELETE("deleteScore", competitionTeamApi.DeleteTeamScore)
	}
	{
		routerWithoutRecord.GET("competitionTeam", competitionTeamApi.GetCompetitionTeam)
		routerWithoutRecord.GET("competitionTeamList", competitionTeamApi.GetCompetitionTeamList)
		routerWithoutRecord.GET("scores", competitionTeamApi.GetTeamScores)
		routerWithoutRecord.GET("recentScores", competitionTeamApi.GetTeamRecentScores)
		routerWithoutRecord.GET("detail", competitionTeamApi.GetTeamDetail)
		routerWithoutRecord.GET("allScores", competitionTeamApi.GetAllTeamsScoreSummary)
		routerWithoutRecord.GET("warTable", competitionTeamApi.GetWarTableDownload)
	}
	// 公开路由（无鉴权）
	publicGroup := PublicRouter.Group("competitionTeam")
	{
		publicGroup.GET("public/warScores", competitionTeamApi.GetPublicWarScores)
		publicGroup.GET("public/warBounty", competitionTeamApi.GetPublicWarBounty)
		publicGroup.GET("public/teamList", competitionTeamApi.GetPublicTeamList)
		publicGroup.GET("public/warTable", competitionTeamApi.GetWarTableDownload)
		publicGroup.GET("ranking", competitionTeamApi.GetTeamScoreRanking)
		publicGroup.GET("bountyRanking", competitionTeamApi.GetTeamBountyRanking)
	}
}
